frontend/src/pages/secret-manager/OverviewPage/components/SecretOverviewTableRow/SecretEditRow.tsx

```diff
@@ -435,6 +412,47 @@ export const SecretEditRow = ({
               </Modal>
             </div>
 
+            <ProjectPermissionCan
+              I={ProjectPermissionActions.Read}
+              a={subject(ProjectPermissionSub.Secrets, {
+                environment,
+                secretPath,
+                secretName,
+                secretTags: ["*"]
+              })}
+            >
+              {(isAllowed) => (
+                <InlineActionIconButton
+                  hint="Share Secret"
+                  icon={faShare}
+                  isDisabled={!isAllowed}
+                  onClick={openCreateSharedSecretPopup}
+                  revealOnGroupHover
+                />
+              )}
+            </ProjectPermissionCan>
+
+            <ProjectPermissionCan
+              I={ProjectPermissionActions.Create}
+              a={subject(ProjectPermissionSub.Secrets, {
+                environment,
+                secretPath,
+                secretName,
+                secretTags: ["*"]
+              })}
+            >
+              {(isAllowed) => (
+                <InlineActionIconButton
+                  hint="Add Override"
+                  isHidden={hasOverride}
+                  icon={faCodeBranch}
+                  isDisabled={!isAllowed}
+                  onClick={handleAddPersonalOverride}
+                  revealOnGroupHover
+                />
+              )}
+            </ProjectPermissionCan>
+
```

Comment:
Permission for creating a secret override and sharing replicated from `SecretItem`